lm_eval/models/ibm_watsonx_ai.py

```diff
@@ -284,126 +256,105 @@ def loglikelihood(self, requests: List[Instance]) -> List[Tuple[float, bool]]:
                 2. a target string on which the loglikelihood of the LM producing this target,
                    conditioned on the input, will be returned.
         Returns:
-            tuple (loglikelihood, is_greedy) for each request according to the input order:
+            Tuple (loglikelihood, is_greedy) for each request according to the input order:
                 loglikelihood: probability of generating the target string conditioned on the input
                 is_greedy: True if and only if the target string would be generated by greedy sampling from the LM
         """
         try:
             from ibm_watsonx_ai.metanames import GenTextParamsMetaNames as GenParams
         except ImportError:
             raise ImportError(
-                "Could not import ibm_watsonx_ai: Please install lm_eval[ibm_watsonx_ai] package."
+                "Could not import ibm_watsonx_ai: Please install ibm_watsonx_ai package."
             )
         self._check_model_logprobs_support()
-        self.generate_params[GenParams.MAX_NEW_TOKENS] = 1
+        generate_params = copy.deepcopy(self.generate_params)
+        generate_params[GenParams.MAX_NEW_TOKENS] = 1
```

Comment:
NIT: I wonder if we could just add `MAX_NEW_TOKENS` to `self.generate_params` and then pop it off after the loop to avoid the deep copy? Of course, with a `dict` of this size, deepcopy probably isn't incurring much additional overhead - just an idea!

Reply:
I think we can do a shallow copy (which is much less expensive), since we only override a single field.

Reply:
I've changed it to be a shallowcopy